### PR TITLE
Define component-based triage meetings

### PR DESCRIPTION
**Pull request by andrross**
### Description
A couple weeks ago in the core triage meeting we discussed scoping the "untriaged" issue search to component-specific areas, giving each respective component meeting the responsibility to triage new issues. I think it is easier to discuss this change via a PR update to the document that defines this process, so I'm doing that here as opposed to creating an issue. I will not merge this change until I get consensus from all component meeting owners (since they will be taking on a new responsibility here).

- [x] Search: @msfroh or @getsaurabh02 
- [ ] Storage: @linuxpi or @rramachand21 or @Bukhtawar 
- [ ] Cluster Manager: @rwali-aws 
- [x] Core: @peternied 

/cc @reta @dblock @shwetathareja @Bukhtawar 

---

We have expanded beyond the single "core" triage meeting, but so far the core meeting has looked at all issues. This PR proposes an update to the procedure so that the component-specific meetings now have the responsibility for addressing untriaged issues for their component.

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [x] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Bukhtawar — COMMENTED** [review]
Thanks Andrew, much needed. Quick question, do we need some guidance to standardise the component triage meetings

**Comment by andrross** [comment]
> Thanks Andrew, much needed. Quick question, do we need some guidance to standardise the component triage meetings

@Bukhtawar Maybe? Let's start with the structure that is defined in the "Is there an agenda for each week?" section currently and update as necessary.

File: TRIAGING.md
Comment (peternied) [inline TRIAGING.md:24]: I think this is important and would be better if it was inline in the agenda so it was easy to follow during the meetings.  Seems like it should sit between the existing items 4 and 5 - what do you think?

File: TRIAGING.md
Comment (peternied) [inline TRIAGING.md:19]: What do you think about moving these search urls, the existing agenda doesn't take these into account on line 41 (bullet 4).  I'd rather these were build into the steps that are followed, otherwise the person running triage needs to remember and reference up in the document.

Could we bake these into that link so the person running triage only needs to click a link according the meeting they are in?

File: TRIAGING.md
Comment (peternied) [inline TRIAGING.md:21]: Can we include the links to the specific meetup series for each of these?

File: TRIAGING.md
Comment (peternied) [inline TRIAGING.md:17]: I don't think attendees are interested in 'why' they just want to know the answer to the question. What do you think of the following?
```suggestion
There are several weekly triage meetings scoped to the following component areas: Search, Storage, Cluster Manager, and finally "Core" as a catch-all for all other issues.  Depending on the issue's labels will determine which triage meeting will review the issue.
```

I think the question `How do I join the Triage meeting?` should still be first, the answer just got a little more interesting with many potential meetings.

Comment (andrross) [reply TRIAGING.md:17]: Totally agree the "why" isn't necessary and I like the shorter text, but I will not be taking your two-spaces-after-the-period suggestion

File: TRIAGING.md
Comment (andrross) [reply TRIAGING.md:19]: Sure, I'll do that

File: TRIAGING.md
Comment (andrross) [reply TRIAGING.md]: I moved this to the agenda section

File: TRIAGING.md
Comment (andrross) [reply TRIAGING.md]: Done

File: TRIAGING.md
Comment (andrross) [reply TRIAGING.md]: I actually moved the "There are several weekly triage meetings..." line to the top paragraph. Let me know what you think.

File: TRIAGING.md
Comment (peternied) [inline TRIAGING.md:31]: Looks like you need another space for these to be picked up by markdown as 'sub-bullets' of item 4, you can see this on the rich diff: 
image

**Review by peternied — APPROVED** [review]

File: TRIAGING.md
Comment (peternied) [inline TRIAGING.md:35]: In the previous iteration you had a finding _any_ older untriaged issues, did you want to put that back in around here?  I think it would be great if every triage meeting double checked one another.

**Review by peternied — APPROVED** [review]

File: TRIAGING.md
Comment (andrross) [reply TRIAGING.md:35]: Yes, good catch

Comment (andrross) [reply TRIAGING.md:35]: I added that text in a 7th agenda item

**Review by peternied — APPROVED** [review]

**Comment by andrross** [comment]
I'd like to get approvals from a representative from each of the component meetings before merging this, since this results in a change to how those meetings will be conducted:

- [ ] Search: @msfroh or @getsaurabh02 
- [ ] Storage: @linuxpi or @rramachand21 or @Bukhtawar 
- [ ] Cluster Manager: @rwali-aws 
- [x] Core: @peternied

**Review by getsaurabh02 — APPROVED** [review]
Thanks Andrew! This was much needed. LGTM.

File: TRIAGING.md
Comment (rwali-aws) [inline TRIAGING.md:3]: sometime we see that issues have more than one component tags. So, based on the proposal, if an issue has one of the above tags and one of catch-all tags, would it be covered in the component meetings or in the both meetings?

**Comment by rramachand21** [comment]
Thanks Andrew - should we also define the steps to "remove" the untriaged label please during the triage meetings? In my opinion, untriaged should only be removed after all the key attributes are filled for an issue. There is not standardization as of today but this would be helpful to be consistent across areas/components.

File: TRIAGING.md
Comment (andrross) [reply TRIAGING.md:3]: I would say that whichever the meeting happens first should look at it, and then do one of two things:
- triage the issue, potentially removing the other component tags if they are not applicable
- remove the component tag the current meeting "owns" and leave the issue untriaged so that the other meeting can triage it

What do you think?

**Comment by andrross** [comment]
> Thanks Andrew - should we also define the steps to "remove" the untriaged label during the triage meetings? In my opinion, untriaged should only be removed after all the key attributes are filled for an issue. There is not standardization as of today but this would be helpful to be consistent across areas/components.

@rramachand21 We do have some guidance in the ["typical outcomes"](https://github.com/opensearch-project/OpenSearch/blob/b59c5726d2cfb4e0a0b2143aa7f8aebb707866fb/TRIAGING.md?plain=1#L66-L73) section. I'm certainly not opposed to adding more details, but I'm thinking we could expand on that in a separate PR. What do you think?

File: TRIAGING.md
Comment (rwali-aws) [reply TRIAGING.md:3]: If I understand correctly, this is what we are ending up doing today. Atleast, thats the case for Cluster Manager.
If the intent is to baseline this across components, then I am good.

However, in the fullness of time, we should see if we can have a mechanism to avoid double triaging. To this, I dont have a good answer at the moment. :genie:

**Review by peternied — APPROVED** [review]

File: TRIAGING.md
Comment (peternied) [reply TRIAGING.md:3]: @rwali-aws thanks for the thoughts - I'm resolving this comment thread to make progress. 

When we get more data points, it would be great to open a PR to update this document for discussion and adoption.

**Comment by peternied** [comment]
@andrross I don't think this needs to be backported, but up to you

**Pull request merged**
